api/features/workflows/core/models.py

```diff
@@ -179,6 +180,30 @@ def _publish_change_sets(self, published_by: "FFAdminUser") -> None:
         for change_set in self.change_sets.all():
             change_set.publish(user=published_by)
 
+    def _publish_segments(self) -> None:
+        segments = list(self.segments.all())
+        if not segments:
+            return
+
+        for segment in segments:
```

Comment:
Is there a reason to do the empty check separately instead of just relying on the `for` loop? e.g. 

```suggestion
        for segment in self.segments.all():
```

Reply:
Nice catch, implemented